solr/core/src/java/org/apache/solr/cluster/placement/plugins/AffinityPlacementFactory.java

```diff
@@ -193,6 +195,22 @@ public PlacementPlan computePlacement(Cluster cluster, PlacementRequest request,
       Set<Node> nodes = request.getTargetNodes();
       SolrCollection solrCollection = request.getCollection();
 
+      final SolrCollection secondaryCollection;
+      String withCollection = solrCollection.getCustomProperty(CollectionAdminParams.WITH_COLLECTION);
```

Comment:
`withCollection` variable here will be the property configured on the collection, it will not resolve the alias if there's one. I believe we need to use params from the placement request to decide if we try to resolve?

Reply:
Taking this back, ignore previous comments. We resolved alias at collection creation time so expecting real collection name here.

Reply:
I think that we should probably completely disallow using aliases in this context... how would that work in practice? if you flip the alias that used to point to the secondary collection to some other collection then suddenly the co-location may be completely broken.